### ISSUE TITLE
New Hammerspoon Icon for 2023?

### ISSUE THREAD

**Issue by latenitefilms** [Hammerspoon/hammerspoon#3355]
@cmsj & @asmagill - Would you have any objections if I paid for a new Hammerspoon icon?

I would get [Matthew Skiles](https://matthewskiles.com) to do it in a similar style to ones he's done for me previously:

![icons](https://user-images.githubusercontent.com/22286696/210311102-6b4ea661-761c-4feb-82eb-12a18ba87638.png)

Thoughts? Objections?

**Comment by dasmurphy** [comment]
The icons look really awesome. I am in for it.

**Comment by latenitefilms** [comment]
@cmsj - Any objections?

**Comment by cmsj** [comment]
@latenitefilms I don't object, but I can't promise I'll like it 😬

**Comment by latenitefilms** [comment]
No worries - I'll press the "go" button and we'll see what happens!

**Comment by luckman212** [comment]
Those icons are very cool. But, I must admit, I'm nostalgic about the current icon and think I would miss it.

**Comment by dasmurphy** [comment]
May be we can implement an icon switcher for the old/new icon as a setting?

**Comment by ameddin73** [comment]
Those are awesome but I'm worried they might scale weird down to favicon size...

**Comment by cmsj** [comment]
I'll mention that I did the current icon and I have absolutely no background in graphic design.

While I stand by my earlier comment that I don't promise to love a newer icon design, I have substantial trust that a professional designer is likely to do a decent job, especially for the smaller scales of icon.

**Comment by latenitefilms** [comment]
To be honest, the main reason I wanted to look into changing it is totally selfish, and was because when I do presentations/talks about the software I work on, the Hammerspoon icon doesn't fit in as nicely along with my other app icons (as seen in the first post) - however, looking at it with fresh eyes this morning, I totally forgot that the Hammerspoon icon was modernised in #2719 a while back, and I haven't been using that revised version for these presentations/talks.

Whilst I don't think the Hammerspoon icon looks quite as modern, as say the Xcode icon - I'm going to close this issue for now, and maybe have another look at it later in the year if funds permit. The only reason I'm willing to put my own money into it is because CommandPost wouldn't exist without Hammerspoon - it's literally the engine that drives it.

If we did change the icon, I don't think it would be a dramatic change - I just like the ideal of the hammer/spoon being a bit more photo-real (like the Xcode icon). Maybe it could even have a bit of a "looking the the hood" art some Lua code similar to Hopper Dissembler? Basically something that gives the same vibe, and is still recognisable as the Hammerspoon icon, but fits in more nicely with Apple's current design styles. 

I also agree with @dasmurphy that if we did this, we should expose some APIs that allow you to change it to whatever - I know @cmsj played around with doing animated dock icons a while back (I can't seem to see the issue via search), and this is something I also have been meaning to play with at some point for CommandPost.
